runtime/runtime-params-estimator/src/action_costs.rs

```diff
@@ -0,0 +1,626 @@
+        .add_action(add_fn_access_key_action(ActionSize::Min))
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn add_function_call_key_base_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(add_fn_access_key_action(ActionSize::Min))
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn add_function_call_key_base_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(add_fn_access_key_action(ActionSize::Min))
+        .inner_iters(1) // adding the same key a second time would fail
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .apply_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn add_function_call_key_byte_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(add_fn_access_key_action(ActionSize::Max))
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+        / ActionSize::Max.key_methods_list()
+}
+
+pub(crate) fn add_function_call_key_byte_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(add_fn_access_key_action(ActionSize::Max))
+        .verify_cost(&mut ctx.testbed())
+        / ActionSize::Max.key_methods_list()
+}
+
+pub(crate) fn add_function_call_key_byte_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(add_fn_access_key_action(ActionSize::Max))
+        .inner_iters(1) // adding the same key a second time would fail
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .apply_cost(&mut ctx.testbed())
+        / ActionSize::Max.key_methods_list()
+}
+
+pub(crate) fn delete_key_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(delete_key_action())
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn delete_key_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx).add_action(delete_key_action()).verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn delete_key_exec(ctx: &mut EstimatorContext) -> GasCost {
+    // Cannot delete a key without creating it first. Therefore, compute cost of
+    // (create) and of (create + delete) and return the difference.
+    let base = ActionEstimation::new(ctx)
+        .add_action(add_fn_access_key_action(ActionSize::Max))
+        .predecessor(AccountRequirement::SameAsSigner)
+        .inner_iters(1)
+        .receiver(AccountRequirement::SameAsSigner)
+        .apply_cost(&mut ctx.testbed());
```

Comment:
Any reason this is open-coded rather than delegating to `add_function_call_key_base_exec`?

Alternatively, this could remain open-coded, but could the builder be reused between this and the next invocation, as in:

```rust
let base_builder = ActionEstimation::new(ctx)
    .add_action(add_fn_access_key_action(ActionSize::Max))
    .predecessor(AccountRequirement::SameAsSigner)
    .inner_iters(1)
    .receiver(AccountRequirement::SameAsSigner);
let base = base_builder.apply_cost(&mut ctx.testbed());
let total = base_builder
    .add_action(delete_key_action())
    .inner_iters(100)
    .apply_cost(&mut ctx.testbed());
```

Reply:
Yes I like the second alternative, it keeps the code local. Updated accordingly.